lxd/device/disk.go

```diff
@@ -2597,11 +2597,24 @@ func (d *disk) generateVMConfigDrive() (string, error) {
 		}
 	}
 
-	// Append any custom meta-data to our predefined meta-data config.
+	reservedUserKeys := []string{"user.network-config", "user.user-data", "user.vendor-data", "user.meta-data"}
+
 	metaData := fmt.Sprintf(`instance-id: %s
-local-hostname: %s
-%s
-`, d.inst.Name(), d.inst.Name(), instanceConfig["user.meta-data"])
+	local-hostname: %s
+`, d.inst.Name(), d.inst.Name())
+
+	// Add other user.* keys to meta-data so that cloud-init has access to this information.
+	for key, value := range instanceConfig {
+		if strings.HasPrefix(key, "user.") && !shared.ValueInSlice(key, reservedUserKeys) {
+			fmt.Println(key)
```

Comment:
is this old debug ?

Reply:
Rather than using fmt.Sprintf in this function, how about using strings.Builder to build up the contents of the file?